### DISCUSSION TITLE
"How can I send the data collected by node_exporter to Pushgateway?"

### DISCUSSION THREAD

**Discussion by donnasky** [prometheus/pushgateway#545] [Q&A]
I have a prometheus server and pushgateway,These two are installed on machines with public IP addresses, while the machine where node_exporter is installed only has broadband network without a public IP address, and it can access Pushgateway and Prometheus(prometheus can not reach node_exporter directly). In this case, how can I send the data collected by node_exporter to Pushgateway? Are there any reference solutions available?Thanks

**Comment by donnasky** [comment]
Port forwarding is one method, but I am more interested in knowing if there is a machine on the node_exporter side that can directly push the data to Pushgateway, which can be done through some middleware or program

**Comment by beorn7** [comment]
The PGW is not designed to turn Prometheus into a push-based monitoring system. Just pushing a whole scrape to the PGW is an anti-pattern (even if you maybe could make it work with some makeshift contraption.

More appropriate approaches in your case might be [PushProx](https://github.com/prometheus-community/PushProx) or utilizing [Prometheus agent mode](https://prometheus.io/blog/2021/11/16/agent/) and the remote-write protocol.
